src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -152,7 +159,11 @@ public void handleException(TransportException exp) {
             try {
                 timedOutWaitingForNodes.set(!publishResponseHandler.awaitAllNodes(publishTimeout));
                 if (timedOutWaitingForNodes.get()) {
-                    logger.debug("timed out waiting for all nodes to process published state [{}] (timeout [{}])", clusterState.version(), publishTimeout);
+                    DiscoveryNode[] pendingNodes = publishResponseHandler.pendingNodes();
```

Comment:
Not sure I follow? my point here was to log the lagging nodes, so one can go to them and research what's going on.